backends-velox/src/main/scala/org/apache/spark/sql/execution/VeloxColumnarToRowExec.scala

```diff
@@ -70,118 +69,106 @@ case class VeloxColumnarToRowExec(child: SparkPlan) extends ColumnarToRowExecBas
     val numOutputRows = longMetric("numOutputRows")
     val numInputBatches = longMetric("numInputBatches")
     val convertTime = longMetric("convertTime")
-
-    new ColumnarToRowRDD(
-      sparkContext,
-      child.executeColumnar(),
-      this.output,
-      numOutputRows,
-      numInputBatches,
-      convertTime)
+    child.executeColumnar().mapPartitions {
+      it =>
+        VeloxColumnarToRowExec.toRowIterator(
+          it,
+          output,
+          numOutputRows,
+          numInputBatches,
+          convertTime
+        )
+    }
   }
 
   protected def withNewChildInternal(newChild: SparkPlan): VeloxColumnarToRowExec =
     copy(child = newChild)
 }
 
-class ColumnarToRowRDD(
-    @transient sc: SparkContext,
-    rdd: RDD[ColumnarBatch],
-    output: Seq[Attribute],
-    numOutputRows: SQLMetric,
-    numInputBatches: SQLMetric,
-    convertTime: SQLMetric)
-  extends RDD[InternalRow](sc, Seq(new OneToOneDependency(rdd))) {
+object VeloxColumnarToRowExec {
```

Comment:
move code to `VeloxColumnarToRowExec.toRowIterator` so that columnar serializer can reuse it